ratpack-manual/src/content/chapters/37-pac4j.md

```diff
@@ -0,0 +1,141 @@
+Thus, configuring the ways to login into your application requires to create the right [clients](https://github.com/pac4j/pac4j/wiki/Clients): a `FacebookClient` to log in with Facebook, a `TwitterClient` to log in with Twitter, a `Google2Client` to log in with Google (using OAuth v2.0 protocol), and so on.
+
+Create your `pac4j` clients and define them **BEFORE** any other authentication operation in your web chain using the [`RatpackPac4j.authenticator(Client<?, ?>... clients)`](api/ratpack/pac4j/RatpackPac4j.html#authenticator-org.pac4j.core.client.Client...-) method. Then secure an url, get the authenticated user profile or log out. 
+
+```language-java
+import com.google.appengine.repackaged.com.google.common.collect.Maps;
+import org.pac4j.cas.client.CasClient;
+import org.pac4j.core.client.Client;
+import org.pac4j.core.client.Clients;
+import org.pac4j.core.context.WebContext;
+import org.pac4j.core.profile.UserProfile;
+import org.pac4j.http.client.BasicAuthClient;
+import org.pac4j.http.client.FormClient;
+import org.pac4j.http.credentials.SimpleTestUsernamePasswordAuthenticator;
+import org.pac4j.http.profile.UsernameProfileCreator;
+import org.pac4j.oauth.client.FacebookClient;
+import org.pac4j.oauth.client.TwitterClient;
+import ratpack.func.Action;
+import ratpack.groovy.template.TextTemplateModule;
+import ratpack.guice.Guice;
+import ratpack.handling.Chain;
+import ratpack.pac4j.RatpackPac4j;
+import ratpack.server.RatpackServer;
+import ratpack.server.ServerConfig;
+import ratpack.session.SessionModule;
+
+import java.io.File;
+import java.util.Map;
+import java.util.Optional;
+
+import static java.util.Collections.singletonMap;
+import static ratpack.groovy.Groovy.groovyTemplate;
+
+public class Example {
+
+    public static void main(final String[] args) throws Exception {
+        RatpackServer.start(server -> server
+                .serverConfig(ServerConfig
+                        .baseDir(new File("src/main"))
+                        .port(8080)
+                )
+                .registry(Guice.registry(b -> b
+                        .module(TextTemplateModule.class)
+                        .module(SessionModule.class)
+                ))
+                .handlers(chain -> {
+                    final FacebookClient facebookClient = new FacebookClient("fkey", "fsecret");
+                    final TwitterClient twitterClient = new TwitterClient("tkey", "tsecret");
+                    final FormClient formClient = new FormClient("/theForm.html", new SimpleTestUsernamePasswordAuthenticator(), new UsernameProfileCreator());
+                    final BasicAuthClient basicAuthClient = new BasicAuthClient(new SimpleTestUsernamePasswordAuthenticator(), new UsernameProfileCreator());
+                    final CasClient casClient = new CasClient();
+                    casClient.setCasLoginUrl("http://mycasserver/login");
```

Comment:
I feel there's too much happening in this example. We don't need to show the usage of every client in this example. It should be kept to:
* How to configure a client and require authentication
* How to logout
* How to obtain and use the userProfile